tests/unit/core/api/test_index.js

```diff
@@ -41,15 +41,45 @@ describe(__filename, () => {
         .then(() => mockWindow.verify());
     });
 
-    it('encodes non-ascii URLs in UTF8', () => {
+    it('does not encode non-ascii URLs in UTF8 on the client', () => {
+      const endpoint = 'project-ă-ă-â-â-日本語';
+      mockWindow.expects('fetch')
+        .withArgs(`${apiHost}/api/v3/${endpoint}/`, {
+          body: undefined, credentials: undefined, method: 'GET', headers: {},
+        })
+        .once()
+        .returns(createApiResponse());
+
+      // We use `cloneDeep()` to allow modifications on the `config` object,
+      // since a call to `get()` makes it immutable.
+      const clientConfig = configUtil.cloneDeep(config);
+      clientConfig.client = true;
+      clientConfig.server = false;
+
+      return api.callApi({
+        _config: clientConfig,
+        endpoint,
+      }).then(() => mockWindow.verify());
+    });
+
+    it('encodes non-ascii URLs in UTF8 on the server', () => {
       const endpoint = 'diccionario-español-venezuela';
       mockWindow.expects('fetch')
         .withArgs(utf8.encode(`${apiHost}/api/v3/${endpoint}/`), {
           body: undefined, credentials: undefined, method: 'GET', headers: {},
         })
         .once()
         .returns(createApiResponse());
-      return api.callApi({ endpoint }).then(() => mockWindow.verify());
+
+      // We use `cloneDeep()` to allow modifications on the `config` object,
```

Comment:
I don't think it's really necessary to keep putting this comment everywhere. It doesn't hurt though.